imsim/diffraction_fft.py

```diff
@@ -0,0 +1,226 @@
+import numpy as np
+import scipy.signal
+
+from .diffraction import field_rotation_sin_cos, prepare_e_z, e_equatorial
+
+# Obtained from linear regression of the photon shooting r distribution:
+INTERCEPT = 0.0706052627908828
+A_SQRT = 1.0 / (2.0 * np.exp(INTERCEPT) * np.pi)
+
+
+def spike_profile(r: np.ndarray) -> np.ndarray:
+    """Spike profile fit to the spikes produced by the statistical approach."""
+    return 0.5 * A_SQRT / np.pi / (1 + (A_SQRT * r) ** 2)
+
+
+def int_spike_profile(r: np.ndarray) -> np.ndarray:
+    """Antiderivative of the spike profile."""
+    return 0.5 * A_SQRT / np.pi / A_SQRT * np.arctan(A_SQRT * r)
+
+
+def field_rotation_profile(
+    r: np.ndarray, d_alpha: float, resolution: float
+) -> np.ndarray:
+    r"""Due to field rotation, the spike cross is rotating with time.
+    As a consequence outer pixels receive a lower dose of photons than inner pixels.
+    This function returns version of `spike_profile` including the field rotation effect.
+
+    Idea: Calculate the photon dose a single pixel in a distance of r to the
+    spike center receives.
+
+    Approximations:
+    - Assume that the rotation rate is constant over time.
+    - A pixel is approximated by an "angular pixel" (r0-1/2 <= r <= r0+1/2, 0 <= \phi <= \phi_pix(r0), where \phi_pix(r) is chosen such that the arclength r \phi_pix(r) is 1.
+    """
+
+    l_pix = resolution  # length of a pixel
+
+    def arclength_dose(r: np.ndarray) -> np.ndarray:
+        """If the arclength r * d_alpha is smaller than the length of a pixel,
+        return 1.0, otherwise return l_pix / (r * d_alpha)."""
+        return 1.0 / np.maximum(r * d_alpha / l_pix, 1.0)
+
+    d_pix = 0.5 * l_pix  # Half a pixel
+    return (
+        int_spike_profile(r + d_pix) - int_spike_profile(r - d_pix)
```

Comment:
I tried using the approximation instead. I actually also was not expecting this, this would result in a difference in the angle of ~ 1° (and a difference in its stddev of ~ 3°). I ll leave the integral version and explain it